File: php/switchcaseAtm.php
<?php
// ===== Program ATM PHP =====

do {
    echo "Masukkan Nama: ";
    $nama = trim(fgets(STDIN));

    echo "Apakah nama sudah benar? (true/false): ";
    $cek = trim(fgets(STDIN));
} while ($cek !== "true");

do {
    echo "Masukkan NIM (angka saja): ";
    $nim = trim(fgets(STDIN));
} while (!ctype_digit($nim));

$saldo = (int)$nim;

echo "\n===== DATA NASABAH =====\n";
echo "Nama: $nama\n";
echo "Saldo Awal: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";

while (true) {
    echo "===== MENU ATM =====\n";
    echo "1. Cek Saldo\n";
    echo "2. Tarik Tunai\n";
    echo "3. Setor Tunai\n";
    echo "4. Transfer\n";
    echo "5. Keluar\n";
    echo "Pilih menu: ";
    $menu = trim(fgets(STDIN));

    switch ($menu) {
        case 1:
            echo "Saldo Anda: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";
            break;

        case 2:
            echo "Masukkan jumlah tarik tunai: ";
            $tarik = trim(fgets(STDIN));
            if ($tarik <= $saldo) {
                $saldo -= $tarik;
                echo "Penarikan berhasil!\n";
            } else {
                echo "Saldo tidak cukup!\n";
            }
            break;

        case 3:
            echo "Masukkan jumlah setor tunai: ";
            $setor = trim(fgets(STDIN));
            $saldo += $setor;
            echo "Setor tunai berhasil!\n";
            break;

        case 4:
            echo "Masukkan jumlah transfer: ";
            $transfer = trim(fgets(STDIN));
            if ($transfer <= $saldo) {
                $saldo -= $transfer;
                echo "Transfer berhasil!\n";
            } else {
                echo "Saldo tidak cukup!\n";
            }
            break;

        case 5:
            echo "Terima kasih telah menggunakan ATM!\n";
            exit;

        default:
            echo "Menu tidak tersedia!\n";
    }
}
?>

File: python/switchcaseATM.py
# ===== Program ATM Python =====

while True:
    nama = input("Masukkan Nama: ")
    cek = input("Apakah nama sudah benar? (true/false): ")
    if cek.lower() == "true":
        break

while True:
    nim = input("Masukkan NIM (angka saja): ")
    if nim.isdigit():
        break

saldo = int(nim)

print("\n===== DATA NASABAH =====")
print(f"Nama: {nama}")
print(f"Saldo Awal: Rp {saldo:,}".replace(",", "."))

while True:
    print("\n===== MENU ATM =====")
    print("1. Cek Saldo")
    print("2. Tarik Tunai")
    print("3. Setor Tunai")
    print("4. Transfer")
    print("5. Keluar")

    menu = input("Pilih menu: ")

    match menu:
        case "1":
            print(f"Saldo Anda: Rp {saldo:,}".replace(",", "."))

        case "2":
            tarik = int(input("Jumlah tarik tunai: "))
            if tarik <= saldo:
                saldo -= tarik
                print("Penarikan berhasil!")
            else:
                print("Saldo tidak cukup!")

        case "3":
            setor = int(input("Jumlah setor tunai: "))
            saldo += setor
            print("Setor tunai berhasil!")

        case "4":
            transfer = int(input("Jumlah transfer: "))
            if transfer <= saldo:
                saldo -= transfer
                print("Transfer berhasil!")
            else:
                print("Saldo tidak cukup!")

        case "5":
            print("Terima kasih telah menggunakan ATM!")
            break

        case _:
            print("Menu tidak tersedia!")
